ansible-3

kruchkov-alexandr opened the pull request:
# Выполнено ДЗ №

 - [x] Основное ДЗ
 - [ ] Задание со *

## В процессе сделано:
- создал новый бранч ansible-3 и сделал чекаут
- добавил новую папку roles
- командой ansible-galaxy init db/app сделал новую структуру
- поработал над переносом тасков. описывать всё не буду - по слайдам все делал.
- все эти подготовки сделал ровно до слайда 24. затем первый коммит
- на 25 слайде увидел ошибку бандл инсталлера, но забил хер на это
- добавил новую папку environments и в ней две директории prod и stage
- скопировал туда инвентори(до сих пор не динамический)
- удалил инвентори с корня ансибл директории
- в дефолтном конфиге ансибла поставил инвентори от стейджа
- добавил папки group_vars для стейджа и прода. переместил туда переменные из app.yml db.yml
- почистил от переменных app.yml db.yml
- создал новый файл all в стейдже и прода environments
- сделал коммит(слайд 38))
- определяю переменную по умолчанию в env
- добавляем новые роли
- добавляю директорию playbooks и переношу туда все нужное
- добавляю папку old для архива
- перенёс весь олд в олд :D
- переместил все файлы/директории, как на слайде 46
- сделал коммит
- улучшаем ансибл конфиг
- пересоздал машинки gcp, прочекал плейбуки
- добавил непонятный файл непонятой роли в гитигнор
- заинсталлил эту роль
- добавил проксирование порта
- работа с ansible vault. Создал файл users.yml , vault.key , credentials.yml 
- сделал коммит перед шифрованием
- зашифровал файлы, всё ок

## Как запустить проект:


## Как проверить работоспособность:


## PR checklist
 - [x] Выставил label с номером домашнего задания
 - [x] Выставил label с темой домашнего задания

kruchkov-alexandr committed:
первый коммит, слайд 24. закончил первичную подготовку ролей

kruchkov-alexandr committed:
второй коммит, слайд 29

kruchkov-alexandr committed:
третий коммит, слайд 46

kruchkov-alexandr committed:
четвертый коммит, слайд 65

kruchkov-alexandr committed:
пятый коммит, прогон перед пр

kruchkov-alexandr committed:
прогон трависа

kruchkov-alexandr committed:
прогон трависа
поправил пути в шаблоне пакера

kruchkov-alexandr committed:
небольшой фикс app (перенос фигурной скобки)

kruchkov-alexandr commented:
предыдущий конфликт решил

grim-yawn reviewed (commented):
.idea лучше в gitignore

grim-yawn commented on ansible/vault.key:
```diff
@@ -0,0 +1 @@
+123456_123456_123456
```
>  Не допускайте хранения этого файла в git

Хорошо бы vault.key удалить

kruchkov-alexandr replied:
Сделано.

grim-yawn commented on terraform/modules/app/outputs.tf:
```diff
@@ -1,3 +1,3 @@
 output "app_external_ip" {
   value = "${google_compute_instance.app.network_interface.0.access_config.0.assigned_nat_ip}"
-}
+}
```
Нет пустой строки в конце файла

kruchkov-alexandr replied:
Добавил

grim-yawn commented on terraform/prod/main.tf:
```diff
@@ -0,0 +1,26 @@
+  source          = "../modules/vpc"
+  public_key_path = "${var.public_key_path}"
+  zone            = "${var.zone}"
+  source_ranges   = ["77.51.127.140/32"]
```
Наверное лучше вынести в переменную

kruchkov-alexandr replied:
оставллю так.

grim-yawn commented on terraform/prod/variables.tf:
```diff
@@ -0,0 +1,36 @@
+  description = "Path to the public key used for ssh access"
+}
+
+variable disk_image {
```
Кажется переменная не используется

kruchkov-alexandr replied:
Исправил. Только лишнее не public key, а private key

grim-yawn commented on terraform/stage/variables.tf:
```diff
@@ -0,0 +1,36 @@
+  description = "Path to the public key used for ssh access"
+}
+
+variable disk_image {
```
Кажется переменная не используется

kruchkov-alexandr replied:
Исправил. Только лишнее не public key, а private key

grim-yawn commented on terraform/variables.tf.backup:
```diff
@@ -0,0 +1,34 @@
+variable project {
```
Похоже на временный файл

kruchkov-alexandr replied:
Нет

grim-yawn commented on terraform/vpc.tf.backup:
```diff
@@ -0,0 +1,9 @@
+resource "google_compute_firewall" "firewall_ssh" {
```
Похоже на временный файл

kruchkov-alexandr replied:
Нет, не временный.

grim-yawn commented on ansible/credentials.yml:
```diff
@@ -0,0 +1,21 @@
+ansible/environments/prod/credentials.yml
```
Лишний файл?

kruchkov-alexandr replied:
Нет, в нем храниться защифрованное ansible vault

kruchkov-alexandr committed:
прогон травис

kruchkov-alexandr committed:
git rm --cached ansible/vault.key
add last blank string at "/terraform/modules/app/outputs.tf"
deleted variable at "terraform/prod/variables.tf"
deleted variable "Path to private key" an "terraform/prod/main.tf"

kruchkov-alexandr committed:
git rm --cached -r .idea

kruchkov-alexandr committed:
Итогово сделано:

git rm --cached ansible/vault.key
add last blank string at "/terraform/modules/app/outputs.tf"
deleted variable at "terraform/prod/variables.tf"
deleted variable "Path to private key" an "terraform/prod/main.tf"

kruchkov-alexandr committed:
Merge branch 'master' into ansible-3

kruchkov-alexandr commented:
Ожидаю разрешение на мердж

kruchkov-alexandr committed:
поправил ридми файл

kruchkov-alexandr committed:
Merge remote-tracking branch 'origin/ansible-3' into ansible-3

# Conflicts:
#	README.md

kruchkov-alexandr committed:
поправил ридми файл
